casa/System/AipsrcVector.tcc

```diff
@@ -112,42 +99,44 @@ Bool AipsrcVector<T>::find(Vector<T> &value, const String &keyword,
 template <class T>
 uInt AipsrcVector<T>::registerRC(const String &keyword,
 				 const Vector<T> &deflt) {
-  uInt n = Aipsrc::registerRC(keyword, myp_p.ntlst);
-  myp_p.tlst.resize(n);
-  find ((myp_p.tlst)[n-1], keyword, deflt);
+  const uInt n = Aipsrc::registerRC(keyword, ntlst);
+  if(n > tlst.size())
+    tlst.resize(n);
+  find ((tlst)[n-1], keyword, deflt);
   return n;
 }
 
 template <class T>
 uInt AipsrcVector<T>::registerRC(const String &keyword,
 				 const Unit &defun, const Unit &resun,
 				 const Vector<T> &deflt) {
-  uInt n = Aipsrc::registerRC(keyword, myp_p.ntlst);
-  myp_p.tlst.resize(n);
-  find ((myp_p.tlst)[n-1], keyword, defun, resun, deflt);
+  const uInt n = Aipsrc::registerRC(keyword, ntlst);
+  if(n > tlst.size())
+    tlst.resize(n);
+  find ((tlst)[n-1], keyword, defun, resun, deflt);
   return n;
 }
 
 template <class T>
-const Vector<T> &AipsrcVector<T>::get(uInt keyword) {
-  AlwaysAssert(keyword > 0 && keyword <= myp_p.tlst.nelements(), AipsError);
-  return (myp_p.tlst)[keyword-1];
+const Vector<T> AipsrcVector<T>::get(uInt keyword) {
+  AlwaysAssert(keyword > 0 && keyword <= tlst.nelements(), AipsError);
+  return (tlst)[keyword-1];
 }
 
 template <class T>
 void AipsrcVector<T>::set(uInt keyword, const Vector<T> &deflt) {
-  AlwaysAssert(keyword > 0 && keyword <= myp_p.tlst.nelements(), AipsError);
-  (myp_p.tlst)[keyword-1].resize(deflt.nelements());
-  (myp_p.tlst)[keyword-1] = deflt;
+  AlwaysAssert(keyword > 0 && keyword <= tlst.nelements(), AipsError);
+  (tlst)[keyword-1].resize(deflt.nelements());
+  (tlst)[keyword-1] = deflt;
```

Comment:
I thought you could remove the `resize()` but I see that `Array::operator=` calls `Vector::assign_conforming` which requires compatible sizes.
```suggestion
  tlst[keyword-1].resize(deflt.nelements());
  tlst[keyword-1] = deflt;
```